packages/orchestration/src/exos/agoric-names-tools.js

```diff
@@ -0,0 +1,66 @@
+import { VowShape } from '@agoric/vow';
+import { E, Far } from '@endo/far';
+import { M } from '@endo/patterns';
+import { BrandShape } from '@agoric/ertp';
+
+const { Fail } = assert;
+
+/**
+ * @import {NameHub} from '@agoric/vats';
+ * @import {AssetInfo} from '@agoric/vats/src/vat-bank';
+ * @import {Remote} from '@agoric/internal';
+ * @import {Vow, VowTools} from '@agoric/vow';
+ * @import {Zone} from '@agoric/zone';
+ */
+
+/**
+ * Perform remote calls to agoricNames in membrane-friendly way. This is an
+ * interim approach until https://github.com/Agoric/agoric-sdk/issues/9541,
+ * https://github.com/Agoric/agoric-sdk/pull/9322, or
+ * https://github.com/Agoric/agoric-sdk/pull/9519
+ *
+ * XXX consider exposing `has`, `entries`, `keys`, `values` from `NameHub`
+ *
+ * @param {Zone} zone
+ * @param {{ agoricNames: Remote<NameHub>; vowTools: VowTools }} powers
+ */
+export const makeAgNamesTools = (
+  zone,
+  { agoricNames, vowTools: { watch } },
+) => {
+  const agNamesTools = zone.exo(
+    'AgoricNamesTools',
+    M.interface('AgoricNamesToolsI', {
+      lookup: M.call().rest(M.arrayOf(M.string())).returns(VowShape),
+      findBrandInVBank: M.call(BrandShape).returns(VowShape),
+    }),
+    {
+      /** @param {...string} args */
+      lookup(...args) {
+        return watch(E(agoricNames).lookup(...args));
+      },
+      /**
+       * @param {Brand<'nat'>} brand
+       * @returns {Vow<AssetInfo>}
+       */
+      findBrandInVBank(brand) {
+        // XXX consider caching and refetching 1x if brand is not found,
```

Comment:
why refetch? why only once?

Reply:
Likely more clear with this refactor: [3cf3fd0](https://github.com/Agoric/agoric-sdk/pull/9551/commits/3cf3fd03ff2f048f3597b31be6a2cc515bc1f382)